include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdlops_v2r4r2.hpp

```diff
@@ -544,14 +544,9 @@ struct GridwiseGemm_bk0mk1_bk0nk1_mn_xdlops_v2r4r2
                 GetCBlockDescriptor_MBlock_MPerBlock_NBlock_NPerBlock();
 
             auto c_block_buf = make_dynamic_buffer<AddressSpaceEnum::Lds>(
-                static_cast<FloatC*>(p_shared_block),
+                reinterpret_cast<FloatC*>(p_shared_block),
```

Comment:
is this casting from void* to FloatC* ? It should use static_cast

Reply:
No, It is casting from FloatAB to FloatC. using static_cast will result in compile error as below.

`error: static_cast from 'signed char *' to 'int *' is not allowed`

Reply:
p_shared_block should have type ```void*``` instead of ```FloatAB*```, if so, static_cast would work